bokeh/cli/core.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import print_function
+
+import time
+try:
+    import urllib.request as urllib2
+except ImportError:
+    import urllib2
+from six.moves import cStringIO as StringIO
+import pandas as pd
+import numpy as np
+import click
+
+from .help_messages import *
+from .utils import (get_input, get_chart_params, get_charts_mapping,
+                    get_data_series, keep_source_input_sync, get_data_from_url)
+from .. import charts as bc
+
+# Define a mapping to connect chart types supported arguments and chart classes
+CHARTS_MAP = get_charts_mapping()
+
+@click.command()
+@click.option('--input', default=None,help=HELP_INPUT)
+@click.option('--output', default='file://cli_output.html', help=HELP_OUTPUT)
+@click.option('--title', default='Bokeh CLI')
+@click.option('--plot_type', default='Line')
+@click.option('--index', default='', help=HELP_INDEX)
+@click.option('--series', default='', help=HELP_SERIES)
+@click.option('--palette')
+@click.option('--buffer', default='f', help=HELP_BUFFER)
+@click.option('--sync_with_source', default=False)
+def cli(input, output, title, plot_type, series, palette,
+        index, buffer, sync_with_source):
+    """Bokeh Command Line Tool is a minimal client to access high level plotting
+    functionality provided by bokeh.charts API.
+
+    Examples:
+
+    >> python bokeh-cli.py --title "My Nice Plot" --series "High,Low,Close"
+    --plot_type "Line" --palette Reds --input sample_data/stocks_data.csv
+
+    >> cat sample_data/stocks_data.csv | python bokeh-cli.py --buffer t
+
+    >> python bokeh-cli.py --help
+    """
+    cli = CLI(input, output, title, plot_type, series, palette,
```

Comment:
Argh, you caught that by 5 minutes! I had this in mind for a long time and was already fixing it as part of the last commit before changing this to ready (mainly because I'm new to click and didn't know how to spell this). I've renamed this in the last commit.